Fix for bugzilla problem 2314

ckoern opened the pull request:
The [bugzilla problem 2314](https://bugzilla-geant4.kek.jp/show_bug.cgi?id=2314) is still present despite the changes up to version 11.0.0, raising a GeomRegNav1004 exception:
```
-------- EEEE ------- G4Exception-START -------- EEEE -------
*** G4Exception : GeomRegNav1004
      issued by : G4RegularNavigation::ComputeStepSkippingEqualMaterials()
G4RegularNavigation::ComputeStepSkippingEqualMaterials()Stuck Track: potential geometry or navigation problem.
        Track stuck, not moving for 25 steps
- at point (0.136942,-0.11583,-4.28004)
        local direction: (-0.478457,-0.719124,-0.503924)

*** Event Must Be Aborted ***
G4Track (0x7fdea7db9960) - track ID = 1, parent ID = 0
 Particle type : gamma - creator process : not available
 Kinetic energy : 100 keV - Momentum direction : (-0,-0,1)
 Step length : 3.42088 mm  - total energy deposit : 0 eV 
 Pre-step point : (-0.0696549,0.234524,-4.54232) - Physical volume : Target (tm2)
 - defined by : Transportation - step status : 1
 Post-step point : (-0.0696549,0.234524,-4.54232) - Physical volume : Target (tm2)
 - defined by : compt - step status : 7
 *** Note: Step information might not be properly updated.

-------- EEEE -------- G4Exception-END --------- EEEE -------

```
It happens quite frequently to have a single zero step at voxel edges, so for voxel data with large regions containing the same material the threshold for killing the track can be reached.
As i mentioned in comment 3 in the bugzilla issue, the zero-step counter has to be reset when a non-zero step has been performed to prevent this from happening.

I have also fixed the mixed indentation and converted the tabs to spaces in the G4RegularNavigation.cc

arceciemat commented:
Could you please send me the code where your change improved the code? (I will treat it as confidential)

arceciemat commented:
Thanks Gabriele. I think I did not express myself correctly: I meant the example code to test that the new code improves the situation.

ckoern commented:
I have modified the `extended/medical/DICOM` example to show the bug and the fix. You can find it in my fork in the `fix-bugzilla-2314-demo`: https://github.com/ckoern/geant4/tree/fix-bugzilla-2314-demo

Most importantly, I have changed the Parameterisation to use `SetSkipEqualMaterials(true)`.
Other than that, I have changed the primary to a 30 keV gamma with an x-y spread and rotated the phantom to have more voxels along the beam and scaled the voxel size down by a factor of 10.

The `GeomRegNav1004` exception should be present after a few 1000 events.

ckoern commented:
@arceciemat Can you give me an update on this? Will this fix be included in the next release?

ckoern commented:
@arceciemat @gcosmo 
I have just seen, that patch 1 was released but this issue was not addressed. Can you please provide any feedback to this?
This bug in various variations is present since 2019 and I tried to get it fixed via 2 reports in bugzilla and this PR. The bugzilla reports have been closed without checking if the applied changes really work and this PR seems to just be ignored.

Sorry if this sounds too harsh, but it is quite frustrating trying to get a easy bug fixed for 2 1/2 years but being ignored and it prevents us from using the upstream versions

arceciemat commented:
My apologies. I have committed it already
